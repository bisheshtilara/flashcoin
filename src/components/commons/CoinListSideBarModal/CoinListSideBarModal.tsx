import React, { ReactNode } from "react";
import ModalContainer from "../ModalContainer";

export interface IActionsListModalProps {
  isOpen: boolean;
  setIsOpen: React.Dispatch<React.SetStateAction<boolean>>;
  title: string;
  handleAction?: any;
  action: string;
  content?: ReactNode;
}

const CoinListSideBarModal: React.FC<IActionsListModalProps> = ({
  isOpen,
  setIsOpen,
  title,
  handleAction,
  action,
  content,
}) => {
  return (
    <ModalContainer
      isOpen={isOpen}
      setIsOpen={setIsOpen}
      title={title}
      action={action}
      buttonColor="bg-blue-400"
      children={content}
      handleAction={handleAction}
    />
  );
};

export default CoinListSideBarModal;
