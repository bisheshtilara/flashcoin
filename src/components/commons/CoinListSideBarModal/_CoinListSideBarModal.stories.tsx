import { ComponentMeta, ComponentStory } from "@storybook/react";
import CoinListSideBarModal, { IActionsListModalProps } from "./CoinListSideBarModal";

export default {
  component: CoinListSideBarModal,
} as ComponentMeta<typeof CoinListSideBarModal>;


const defaultArgs: IActionsListModalProps = {
  isOpen: true,
  setIsOpen: () => alert("clicked"),
  title: "Confirm",
  handleAction: () => alert("action executed"),
  action: "Confirm",
  content: <p>This is a confirm modal</p>,
};

const Template: ComponentStory<typeof CoinListSideBarModal> = (args: IActionsListModalProps) => (
  <CoinListSideBarModal {...args} />
);

export const Default = Template.bind({});
Default.args = {
  ...defaultArgs,
};

