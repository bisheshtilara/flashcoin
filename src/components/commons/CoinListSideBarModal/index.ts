import CoinListSideBarModal from "./CoinListSideBarModal";
export default CoinListSideBarModal;