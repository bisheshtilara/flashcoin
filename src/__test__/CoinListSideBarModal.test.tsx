import { render, fireEvent, screen } from "@testing-library/react";
import CoinListSideBarModal from "../components/commons/CoinListSideBarModal";

describe("CoinListSideBarModal", () => {
    const setIsOpen = jest.fn();
    it("should render correctly", () => {
        const { container } = render(<CoinListSideBarModal isOpen setIsOpen={setIsOpen} title='Test' action="Confirm"/>);
        expect(container).toMatchSnapshot();
    });
    
    it("should render a modal have title Create test", () => {
        const { container } = render(<CoinListSideBarModal isOpen setIsOpen={() => setIsOpen} title='Create test' action="Confirm"/>);
        // expect(screen.getAllByText("Create test")).toHaveLength(1);
        const cancelButton = screen.getAllByRole("button");
        expect(cancelButton).toHaveLength(2);
    });
});