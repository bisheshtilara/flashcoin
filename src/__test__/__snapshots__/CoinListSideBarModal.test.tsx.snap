// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoinListSideBarModal should render correctly 1`] = `
<div>
  <div
    class="block"
  >
    <div
      class="relative z-10"
    >
      <div
        class="fixed inset-0 bg-gray-500 bg-opacity-75 transition-opacity"
      />
    </div>
    <div
      class="fixed inset-0 z-20 overflow-y-auto"
    >
      <div
        class="flex min-h-full justify-center items-center"
      >
        <div
          class="bg-white  flex flex-col justify-between w-[50%] rounded-lg"
        >
          <div
            class="flex items-center space-x-2 font-semibold border-b p-5"
          >
            <span
              class="text-2xl font-semibold"
            >
              Test
            </span>
          </div>
          <div
            class="p-10"
          />
          <div
            class="flex border-t p-2 space-x-2 justify-end"
          >
            <button
              class="border p-2 w-fit rounded-lg bg-blue-400 text-white  hover:-translate-y-2 duration-300"
            >
              Confirm
            </button>
            <button
              class="border p-2 w-fit rounded-lg bg-slate-400 text-white  hover:-translate-y-2 duration-300"
            >
              Cancel
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
